lib/src/main/java/org/team1540/rooster/drive/pipeline/DriveData.java

```diff
@@ -0,0 +1,85 @@
+package org.team1540.rooster.drive.pipeline;
+
+import java.util.Objects;
+import java.util.OptionalDouble;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Encapsulates data for one side of a tank drivetrain.
```

Comment:
I do like encapsulation, but it seems like this adds a lot of overhead for each packet compared to something like a bunch of streams. Is this worth?

Reply:
Streams would be a lot more complicated API-wise and end-user wise (remember, this framework is designed to be extensible)

Reply:
👌